contracts/NFT.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.20;
+
+import {ERC721} from "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {ReentrancyGuard} from "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+
+/**
+ * @title NFT
+ * @dev This contract allows users to mint ERC721 tokens using ERC20 tokens as collateral.
+ */
+contract NFT is ERC721, Ownable, ReentrancyGuard {
+    address public erc20Address;
+
+    // Maps ERC20 amounts to corresponding token IDs.
+    mapping(uint256 => uint256) public tokenIdsByERC20Amount;
```

Comment:
I think `tokenIdsByERC20Amount` should be a mapping of bool because it makes no sense to store the value twice(as key and as value)